android/src/main/kotlin/com/isvisoft/flutter_screen_recording/FlutterScreenRecordingPlugin.kt

```diff
@@ -116,63 +112,64 @@ class FlutterScreenRecordingPlugin(
         }
     }
 
-    fun calculeResolution(metrics: DisplayMetrics) {
+    fun calculeResolution(screenSize:Point) {
 
-        var maxRes = 1280.0;
+        val screenRatio : Double = (screenSize.x.toDouble()  / screenSize.y.toDouble())
 
-        if (metrics.scaledDensity >= 3.0f) {
-            maxRes = 1920.0;
-        }
+        println(screenSize.x.toString() + " --- " + screenSize.y.toString())
+        var height : Double = mDisplayWidth / screenRatio;
+        println("height - " + height)
 
-        if (metrics.widthPixels > metrics.heightPixels) {
-            val rate = metrics.widthPixels / maxRes
-            mDisplayWidth = maxRes.toInt()
-            mDisplayHeight = (metrics.heightPixels / rate).toInt()
-        } else {
-            val rate = metrics.heightPixels / maxRes
-            mDisplayHeight = maxRes.toInt()
-            mDisplayWidth = (metrics.widthPixels / rate).toInt()
-        }
+        mDisplayHeight = height.toInt()
+
+/*        mDisplayWidth = 2560;
+        mDisplayHeight = 1440;*/
 
         println("Scaled Density")
-        println(metrics.scaledDensity)
+        //println(metrics.scaledDensity)
         println("Original Resolution ")
-        println(metrics.widthPixels.toString() + " x " + metrics.heightPixels)
+        //println(metrics.widthPixels.toString() + " x " + metrics.heightPixels)
         println("Calcule Resolution ")
         println("$mDisplayWidth x $mDisplayHeight")
     }
 
-    fun startRecordScreen() {
-        try {
-            mMediaRecorder?.setVideoSource(MediaRecorder.VideoSource.SURFACE)
-            if (recordAudio!!) {
-                mMediaRecorder?.setAudioSource(MediaRecorder.AudioSource.MIC);
-                mMediaRecorder?.setOutputFormat(MediaRecorder.OutputFormat.THREE_GPP);
-                mMediaRecorder?.setAudioEncoder(MediaRecorder.AudioEncoder.AMR_NB);
-            } else {
-                mMediaRecorder?.setOutputFormat(MediaRecorder.OutputFormat.MPEG_4)
-            }
-            mMediaRecorder?.setOutputFile("${storePath}${videoName}.mp4")
+    fun initMediaRecorder() {
+        mMediaRecorder?.setAudioSource(MediaRecorder.AudioSource.VOICE_RECOGNITION);
+        mMediaRecorder?.setVideoSource(MediaRecorder.VideoSource.SURFACE)
 
-            val rotation = (registrar.activity().getSystemService(Context.WINDOW_SERVICE) as WindowManager).defaultDisplay.orientation
-            mMediaRecorder?.setOrientationHint(rotation * 90)
+        //mMediaRecorder?.setOutputFormat(MediaRecorder.OutputFormat.MPEG_4)
+        mMediaRecorder?.setOutputFormat(MediaRecorder.OutputFormat.THREE_GPP);
+        mMediaRecorder?.setAudioEncoder(MediaRecorder.AudioEncoder.AAC);//AAC //HE_AAC
+        mMediaRecorder?.setAudioEncodingBitRate(16*44100);
```

Comment:
improved audio qulity. all of this section is made step by step to apply with the right order (by a Stackoverflow answer, there seem to be a right order to call the MediaRecorder actions.)